src/template_finder.py

```diff
@@ -161,26 +168,32 @@ def search(self, ref: Union[str, np.ndarray], inp_img: np.ndarray, threshold: fl
         inp_img = inp_img[ry:ry + rh, rx:rx + rw]
 
         if type(ref) == str:
-            template = self._templates[ref][0]
-            scale = self._templates[ref][1]
+            templates = [self._templates[ref][0]]
+            scales = [self._templates[ref][1]]
+        elif type(ref) == list:
+            templates = [self._templates[i][0] for i in ref]
+            scales = [self._templates[i][1] for i in ref]
         else:
-            template = ref
-            scale = 1.0
+            templates = [ref]
+            scales = [1.0]
 
-        img: np.ndarray = cv2.resize(inp_img, None, fx=scale, fy=scale, interpolation=cv2.INTER_NEAREST)
-        rx *= scale
-        ry *= scale
-        rw *= scale
-        rh *= scale
+        for count, template in enumerate(templates):
+            scale = scales[count]
 
-        if img.shape[0] > template.shape[0] and img.shape[1] > template.shape[1]:
-            self.last_res = cv2.matchTemplate(img, template, cv2.TM_CCOEFF_NORMED)
-            _, max_val, _, max_pos = cv2.minMaxLoc(self.last_res)
-            self.last_score = max_val
-            if max_val > threshold:
-                ref_point = (max_pos[0] + int(template.shape[1] * 0.5) + rx, max_pos[1] + int(template.shape[0] * 0.5) + ry)
-                ref_point = (int(ref_point[0] * (1.0 / scale)), int(ref_point[1] * (1.0 / scale)))
-                return True, ref_point
+            img: np.ndarray = cv2.resize(inp_img, None, fx=scale, fy=scale, interpolation=cv2.INTER_NEAREST)
+            rx *= scale
+            ry *= scale
+            rw *= scale
+            rh *= scale
+
+            if img.shape[0] > template.shape[0] and img.shape[1] > template.shape[1]:
+                self.last_res = cv2.matchTemplate(img, template, cv2.TM_CCOEFF_NORMED)
+                _, max_val, _, max_pos = cv2.minMaxLoc(self.last_res)
+                self.last_score = max_val
+                if max_val > threshold:
+                    ref_point = (max_pos[0] + int(template.shape[1] * 0.5) + rx, max_pos[1] + int(template.shape[0] * 0.5) + ry)
+                    ref_point = (int(ref_point[0] * (1.0 / scale)), int(ref_point[1] * (1.0 / scale)))
+                    return True, ref_point
```

Comment:
this changes seem to be more template finder specific, but are not 100% needed for the belt manager. Since the belt manager is quite a big change, lets keep it to it and split this change into another PR?